File: C#/Sort/InsertSort.cs
//insert sort

public class Solution {
   public static int[] InsertSort(int[] nums) {

        var numsLength=nums.Length;

        for(int i=1; i<numsLength; i++)
        {
            var key = nums[i];

            int j = i-1;


            while(j >= 0 && nums[j] > key)
            {
                nums[j+1] = nums[j];
                j-=1;
			}

			nums[j+1]=key;

        }

        return nums;

    }
}

File: Javascript/Sort/InsertSort.js
// insert sort

var insert = function (array, rightIndex, value) {

    var key;

    for (key = rightIndex; key > -1 && array[key] > value; key--) {
        array[key + 1] = array[key];
    }
    array[key + 1] = value;
};

var insertionSort = function (array) {

    for (var i = 1; i < array.length; i++) {
        insert(array, i - 1, array[i]);
    }
};
